src/baseConn.ts

```diff
@@ -774,42 +819,331 @@ export class BaseRawConnection {
     }
 
     if (reExec) {
-      this._validateFetchCardinality(newCard!, asJson, requiredOne);
-      if (this.isLegacyProtocol) {
-        return await this._executeFlow(args, inCodec, outCodec, result);
+      this._validateFetchCardinality(
+        newCard!,
+        outputFormat,
+        expectedCardinality
+      );
+
+      return await this._legacyExecuteFlow(args, inCodec, outCodec, result);
+    }
+  }
+
+  private _encodeParseParams(
+    wb: WriteMessageBuffer,
+    query: string,
+    outputFormat: OutputFormat,
+    expectedCardinality: Cardinality,
+    state: Session,
+    privilegedMode: boolean,
+    options: QueryOptions | undefined
+  ) {
+    wb.writeFlags(
+      0xffff_ffff,
+      privilegedMode ? Capabilities.ALL : RESTRICTED_CAPABILITIES
+    );
+    wb.writeFlags(
+      0,
+      0 |
+        (options?.injectObjectids
+          ? CompilationFlag.INJECT_OUTPUT_OBJECT_IDS
+          : 0) |
+        (options?.injectTypeids ? CompilationFlag.INJECT_OUTPUT_TYPE_IDS : 0) |
+        (options?.injectTypenames
+          ? CompilationFlag.INJECT_OUTPUT_TYPE_NAMES
+          : 0)
+    );
+    wb.writeBigInt64(options?.implicitLimit ?? BigInt(0));
+    wb.writeChar(outputFormat);
+    wb.writeChar(
+      expectedCardinality === Cardinality.ONE ||
+        expectedCardinality === Cardinality.AT_MOST_ONE
+        ? Cardinality.AT_MOST_ONE
+        : Cardinality.MANY
+    );
+    wb.writeString(query);
+
+    if (state === Session.defaults()) {
+      wb.writeBuffer(NULL_CODEC.tidBuffer);
+      wb.writeInt32(0);
+    } else {
+      wb.writeBuffer(this.stateCodec.tidBuffer);
+      if (this.stateCodec === INVALID_CODEC) {
+        wb.writeInt32(0);
```

Comment:
I'm not sure if this is correct - the server will likely fail to decode this zero-lengthed state bytestring.